core/src/main/java/uk/gov/gchq/koryphe/serialisation/json/SimpleClassNameCache.java

```diff
@@ -66,14 +67,14 @@
 public final class SimpleClassNameCache {
     public static final boolean DEFAULT_USE_FULL_NAME_FOR_SERIALISATION = true;
 
-    private static final Set<Class> DEFAULT_PARENT_CLASSES = Set.of(
+    private static final Set<Class> DEFAULT_PARENT_CLASSES = JavaUtils.Set.of(
```

Comment:
Consider using Collections.unmodifiableSet()

Reply:
Internally `JavaUtils.java` is using [`com.google.common.collect.Sets.newHashSet`](https://guava.dev/releases/22.0/api/docs/com/google/common/collect/Sets.html#newHashSet-E...-) which takes varargs. I don't think `Collections.unmodifiableSet()` would work here as it doesn't take varargs.

Reply:
You can do `Collections.unmodifiableSet(JavaUtils.Set.of(...))`...

Reply:
You could, but `Set.of()` ought to always be unmodifiable already. See my other comment.